charts/azuremonitor-containers/templates/ama-logs-deployment.yaml

```diff
@@ -42,6 +43,56 @@ spec:
        - name: TOKEN_NAMESPACE
          value: "azure-arc"
 {{-  .Values.Azure.Identity.MSIAdapterYaml | nindent 7 }}
+     {{- else }}
+     - name: msi-adapter
+       env:
+       - name: TOKEN_NAMESPACE
+         value: azure-arc
+       - name: CLUSTER_IDENTITY
+         value: "false"
+       - name: CLUSTER_TYPE
+         value: {{ (split "/" .Values.Azure.Cluster.ResourceId)._7 }}
+       - name: EXTENSION_ARMID
+         value: {{ .Values.Azure.Extension.ResourceId }}
+       - name: EXTENSION_NAME
+         value: {{ .Values.Azure.Extension.Name }}
+       - name: MSI_ADAPTER_LISTENING_PORT
+         value: "8421"
+       - name: MANAGED_IDENTITY_AUTH
+         value: "true"
+       - name: MSI_ADAPTER_LIVENESS_PORT
+         value: "9090"
+       - name: TEST_MODE
+         value: "false"
+       - name: TEST_FILE
+         value: /data/token
+       image: mcr.microsoft.com/azurearck8s/msi-adapter:1.29.3
```

Comment:
do we need to hard code this image version? can you please find out from Arc k8s team how they handle this image for other clouds in particularly for mooncake?

Reply:
yes https://dev.azure.com/ClusterConfigurationAgent/ClusterConfigurationAgent/_git/ClusterConfigurationAgent?path=/charts/azure-k8s-billing/templates/billing-operator-deployment.yaml&version=GBmain&_a=contents